pkg/controller/reconciler/flux_controller.go

```diff
@@ -0,0 +1,441 @@
+		providerConfig := sdkclients.GenerateProviderConfig(resourceGroup, "", "")
+		marshalledProviderConfig, err := json.MarshalIndent(providerConfig, "", "  ")
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+
+		// Create the namespace if it doesn't exist
+		namespaceObj := &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: namespace,
+			},
+		}
+
+		if err := r.Client.Create(ctx, namespaceObj); err != nil {
+			if !k8serrors.IsAlreadyExists(err) {
+				logger.Error(err, "unable to create namespace")
+				return ctrl.Result{}, err
+			}
+		}
+
+		// Now we should create (or update) each DeploymentTemplate for the bicep files
+		// specified in the git repository.
+		logger.Info("Creating or updating DeploymentTemplate", "name", bicepFile.Name)
+		parameters := convertFromARMJSONParameters(armJSONParameters)
+		err = r.createOrUpdateDeploymentTemplate(ctx, bicepFile.Name, namespace, template, string(marshalledProviderConfig), parameters, repository.Name)
+		if err != nil {
+			logger.Error(err, "failed to create or update deployment template")
+			return ctrl.Result{}, err
+		}
+
+		logger.Info("Successfully created or updated DeploymentTemplate", "name", bicepFile.Name)
+	}
+
+	// List all DeploymentTemplates on the cluster that are from the same git repository
+	deploymentTemplates := &radappiov1alpha3.DeploymentTemplateList{}
+	err = r.Client.List(ctx, deploymentTemplates, client.MatchingFields{deploymentTemplateRepositoryField: repository.Name}, client.InNamespace(""))
+	if err != nil {
+		logger.Error(err, "unable to list deployment templates")
+		return ctrl.Result{}, err
+	}
+
+	logger.Info("Found DeploymentTemplates", "count", len(deploymentTemplates.Items))
+
+	// For all of the DeploymentTemplates on the cluster, check if the bicep file
+	// that it was created from is still present in config. If not, delete the DeploymentTemplate.
+	for _, deploymentTemplate := range deploymentTemplates.Items {
+		if !isSpecifiedInConfig(deploymentTemplate.Name, radiusConfig.Config) {
+			// The DeploymentTemplate is not specified in the config, so we should delete it
+			logger.Info("Deleting DeploymentTemplate", "name", deploymentTemplate.Name)
+			if err := r.Client.Delete(ctx, &deploymentTemplate); err != nil {
+				logger.Error(err, "unable to delete deployment template")
+				return ctrl.Result{}, err
+			}
+
+			logger.Info("Deleted DeploymentTemplate", "name", deploymentTemplate.Name)
+		}
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func (r *FluxController) runBicepBuild(ctx context.Context, filepath, filename string) (armJSON string, err error) {
+	logger := ucplog.FromContextOrDiscard(ctx)
+
+	bicepFile := path.Join(filepath, filename)
+	outFile := path.Join(filepath, strings.ReplaceAll(filename, ".bicep", ".json"))
+
+	// Run bicep restore --force
+	// This is to prevent an issue where bicep build fails to restore the modules
+	logger.Info(fmt.Sprintf("Running bicep restore %s --force", bicepFile))
```

Comment:
do we still need this with the added ssl cert in the dockerfile?

Reply:
nope we can remove it